business/layer.go

```diff
@@ -44,9 +44,20 @@ var kialiCache cache.KialiCache
 
 func initKialiCache() {
 	if config.Get().KubernetesConfig.CacheEnabled {
+		log.Infof("Initializing Kiali Cache")
 		if cache, err := cache.NewKialiCache(); err != nil {
 			log.Errorf("Error initializing Kiali Cache. Details: %s", err)
 		} else {
+			// Temporal NamespaceService to initialize the cache informers
+			initK8sClient := cache.GetClient()
+			initNamespaceService := NewNamespaceService(initK8sClient)
+			if nss, err := initNamespaceService.GetNamespaces(); err != nil {
```

Comment:
This might be out of scope for this change but do we need to create namespace specific caches if `config.Deployment.AccessibleNamespaces` is `**`?

Could the namespace names can come from `config.Deployment.AccessibleNamespaces` instead of the namespace service saving a bunch of kube API calls to fetch the full namespace object if there are many namespaces?

Reply:
> This might be out of scope for this change but do we need to create namespace specific caches if config.Deployment.AccessibleNamespaces is **?

This is exactly my comment in https://github.com/kiali/kiali/issues/4502#issuecomment-1021162055 a potential optimization about this logic is to check/consider that the user/admin may grant cluster permissions to the Kiali's ServiceAccount.

In that case, instead, to create a separated informer per namespace+API, we can create a single informer per API monitoring all namespaces and that would really save memory and network, but again, it should be an optimization that will only work under that case, otherwise, with no cluster right the all-namespaces query will fail and this design should be used.

But yes, it was something that could be introduced and it can be a flag to consider for performance "you need to handle hundreds of namespaces like other Kubernetes admin tool? -> set cluster rights to the ServiceAccount".